.github/ci-scripts/bash/functions-for-build

```diff
@@ -565,6 +591,61 @@ setup_qa_env() {
   deactivate
 }
 
+setup_packer_env() {
+  sudo python3 -m pip uninstall -y packer_framework
+  python3 -m pip install --user virtualenv --trusted-host ${NEXUS_SERVER_ENDPOINT}
+  python3 -m venv pkenv
+  source pkenv/bin/activate
+  python3 -m pip install wheel --trusted-host ${NEXUS_SERVER_ENDPOINT}
+  python3 -m pip install -r ./requirements.txt --extra-index-url=https://$GITHUB_USER:$GITHUB_TOKEN@nexus.tessell.cloud/repository/${NEXUS_PULL_REPOS_PY}/simple
+}
+
+software_def_build() {
+  setupPackerEnv
+  export CURRENT_VERSION_TAG="0.0.0"
+  mkdir -p ~/.packer.d/plugins
+  if [[ "$BRANCH_NAME" == *double_commit* ]]; then
+    echo "BRANCH_NAME contains double_commit"
+    get_pr_commits
+    if [[ "$PR_COMMITS" -le 2 ]]; then
+      echo "BRANCH_NAME contains double_commit and less than 2 commits, skipping build"
+      return
+    fi
+  fi
+  validate_all -r $PWD --src_branch ${SOURCE_BRANCH} --pipeline_run
+  build_all_artifacts -r $PWD --dest_branch ${GITHUB_BASE_REF} --src_branch ${SOURCE_BRANCH} --dry_run --pipeline_run
+}
+
+software_def_build_and_push() {
+  setupPackerEnv
+  export CURRENT_VERSION_TAG=${LATEST_TAG}
+  echo $CURRENT_VERSION_TAG
+  mkdir -p ~/.packer.d/plugins
+  if [[ "$BRANCH_NAME" == *double_commit* ]]; then
+    echo "BRANCH_NAME contains double_commit"
+    get_pr_commits
+    if [[ "$PR_COMMITS" -le 2 ]]; then
+      get_dc_messages
```

Comment:
1. Write a method `get_root_pr_tag` that will abstract all your logic here, and it will be more readable.
2. Won't it be easier and more correct to use the branch name and find its root branch like we discussed? like - `root_branch=${current_branch%_*_double_commit}`?

Reply:
> Write a method get_root_pr_tag that will abstract all your logic here, and it will be more readable.

sure

> for point 2 


I don't think github allows a get-req for a PR based on branch name 
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request

(also when a branch is merged then it's deleted, so I don't think we can get PR information from the deleted branch (I guess for that we'll have to do a get req on number of closed PRs and check if the PR contains the branch info)

Reply:
Also the line which is getting the root pr number from message is this
`root_pr_number=$(echo $dc_message | grep -oE '[0-9]+' | tail -1 )`

which perform the above operation like this 👇 

```
 x="TDEVOPS-2568 | soft lib def rel177 #345  (#361)"     

ci-test-repo-b on  TDEVOPS-2568_soft_lib_def_normal2_main via 🐍 pyenv (packer-documented-test2) on ☁️  (ap-south-1) 
❯ root_pr_number=$(echo $x | grep -oE '[0-9]+' | tail -1 )

ci-test-repo-b on  TDEVOPS-2568_soft_lib_def_normal2_main via 🐍 pyenv (packer-documented-test2) on ☁️  (ap-south-1) 
❯ echo $root_pr_number                                    
361

ci-test-repo-b on  TDEVOPS-2568_soft_lib_def_normal2_main via 🐍 pyenv (packer-documented-test2) on ☁️  (ap-south-1) 
❯ echo $x | grep -oE '[0-9]+'
2568
177
345
361
```